[Lian Zhi Xuan] iP

TehOPanas opened the pull request:
# DukePro 
> “Your mind is for having ideas, not holding them.” - David Allen [(source)](https://dansilvestre.com/productivity-quotes)

+ text-based 
+ ~~Hard~~ Easy to use
+ **SUPER** fast

### All you need to do is :

1. download [here](https://github.com/TehOPanas/ip/releases/tag/Level-9)
2. double-click it
3. enter your tasks
4. let chewie manage for you !!

more importantly .... it's **FREE!**

### Features :

+ [x] Manage your task
+ [ ] Manage your deadline
+ [ ] Reminder

__________________________________________________________

If you're Java programmer, feel free to practice your Java skills too !! Here's the main method :

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

TehOPanas committed:
Implemented starting and ending messages.

TehOPanas committed:
Implemented input and echo.

TehOPanas committed:
Implemented Task Class, createTask and readList feature.

TehOPanas committed:
implemented mark as done feature

TehOPanas committed:
Implemented subclass of task : event, todo, deadline and integrate it into task list

TehOPanas committed:
Implemented testing

TehOPanas committed:
Implemented exception handling.

TehOPanas committed:
implemented delete feature

TehOPanas committed:
Merge remote-tracking branch 'origin/add-gradle-support'

TehOPanas committed:
Add saving feature to Duke, with the use of GSON and import gradle.

TehOPanas committed:
Merge branch 'Level-7'

TehOPanas committed:
Add LocalDate feature

TehOPanas committed:
Merge branch 'Level-8'

TehOPanas committed:
Implement OOP

TehOPanas committed:
Split the files into packages

TehOPanas committed:
Implements unit testing

TehOPanas committed:
Write JavaDoc and improve code style

TehOPanas committed:
Implements find

jeffrey-jian commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,187 @@
+    public void createTaskPrompt (Task task) {
+
+        System.out.println(drawLine());
+        System.out.println("Chewie gotcha, task added:\n" + task.status() + task.taskName());
```
love the idea of your Chewie bot theme!

jeffrey-jian commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,187 @@
+public class Ui {
+
+    //singleton
+    public static Ui ui = new Ui();
```
Honestly am nit-picking here, but perhaps you can rename the singleton `ui` instance as `instance`? To avoid confusion when you call `Ui.ui.method`

jeffrey-jian commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    public List<Task> list() {
+        return list;
+    }
```
```suggestion
    public List<Task> getList() {
        return list;
    }
```

suggestion: rename this getter to `getList`, to improve readability when you call `list.list().othermethod`

jeffrey-jian commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+            if (j < 1 || j > listSize)
+                throw new DukeException("The list doesn't have this index.");
+
+            listSize--;
```
Perhaps instead of using another variable to keep track of the `listSize`, call `list.list().size()` everytime instead, to reduce chances of incrementing/decrementing `listSize` wrongly.

jeffrey-jian reviewed (commented):
Very well thought-out chatbot design, LGTM!! 👍 🏅

Nid21cs commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,102 @@
+ * Task list of user's task
+ *
+ */
+public class TaskList {
```
Plural form could be used to represent a collection of objects

Nid21cs commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,39 @@
+package duke.command;
```
packages are named appropriately

Nid21cs commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
could rename boolean methods to isEquals

Nid21cs commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,151 @@
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+
+import java.io.*;
```
The imported classes should be stated explicitly

TehOPanas committed:
Implemeted GUI and checkstyle.

TehOPanas committed:
Merge branch 'Level-10'

TehOPanas committed:
Modify dialog box

TehOPanas committed:
Add FXML implementation to GUI.

Added MainWindow.java Main.java to
integrate the scene builder with current project.

Modified DialogBox.java to work with FXML.

It is to make designing interface easier and less time consuming.

TehOPanas committed:
Improve code quality.

Rename method name to be more accurate.

Improve readability.

TehOPanas committed:
Change method name in task package.

Modify getter name to fit coding standard.

Improve readability

TehOPanas committed:
Merge branch 'refs/heads/A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/UnmarkCommand.java

TehOPanas committed:
Clean code in Ui.java

TehOPanas committed:
Merge branch 'refs/heads/A-JavaDoc'
Fulfill branch requirement for iP

TehOPanas committed:
Remove unused method

TehOPanas committed:
Merge branch 'refs/heads/Level-9'

TehOPanas committed:
Add assertions

Assert conditions that should be true in parts of code

Make sure that program logic is working as intended

TehOPanas committed:
Merge branch 'A-Assertions'

TehOPanas committed:
Improve code quality

Remove redundant code and simplify methods according to SLAP

Improves readability of code.

TehOPanas committed:
Merge pull request #1 from TehOPanas/A-CodeQuality

Improve code quality

TehOPanas committed:
Implement tagging for task

Each task can have tag assigned to it
It can be searched, added and removed.

This is to further categorise tasks as user wish.

TehOPanas committed:
Merge branch 'BCD-Extension'

TehOPanas committed:
Update JavaDoc and user guide.

TehOPanas committed:
Update Junit test, scale dialog box to text, update task equal method

Update Junit test and equal() method of Task to check
if tags are equal

Update DialogBox to resize according to list size

TehOPanas committed:
Update Junit test and Readme.md

TehOPanas committed:
Modify TerminateCommand to terminate program, fix save bug

TehOPanas committed:
Modify save file's filepath
